packages/flutter/test/material/data_table_test.dart

```diff
@@ -1495,4 +1495,93 @@ void main() {
       const Offset(width - borderVertical, height - borderHorizontal),
     );
   });
+
+  testWidgets('checkboxHorizontalMargin properly applied', (WidgetTester tester) async {
+    const double _customCheckboxHorizontalMargin = 15.0;
+    const double _customHorizontalMargin = 10.0;
+    Finder cellContent;
+    Finder checkbox;
+    Finder padding;
+
+    Widget buildCustomTable({
+      int? sortColumnIndex,
+      bool sortAscending = true,
+      double? horizontalMargin,
+      double? checkboxHorizontalMargin,
+    }) {
+      return DataTable(
+        sortColumnIndex: sortColumnIndex,
+        sortAscending: sortAscending,
+        onSelectAll: (bool? value) {},
+        horizontalMargin: horizontalMargin,
+        checkboxHorizontalMargin: checkboxHorizontalMargin,
+        columns: <DataColumn>[
+          const DataColumn(
+            label: Text('Name'),
+            tooltip: 'Name',
+          ),
+          DataColumn(
+            label: const Text('Calories'),
+            tooltip: 'Calories',
+            numeric: true,
+            onSort: (int columnIndex, bool ascending) {},
+          ),
+          DataColumn(
+            label: const Text('Fat'),
+            tooltip: 'Fat',
+            numeric: true,
+            onSort: (int columnIndex, bool ascending) {},
+          ),
+        ],
+        rows: kDesserts.map<DataRow>((Dessert dessert) {
+          return DataRow(
+            key: ValueKey<String>(dessert.name),
+            onSelectChanged: (bool? selected) {},
+            cells: <DataCell>[
+              DataCell(
+                Text(dessert.name),
+              ),
+              DataCell(
+                Text('${dessert.calories}'),
+                showEditIcon: true,
+                onTap: () {},
+              ),
+              DataCell(
+                Text('${dessert.fat}'),
+                showEditIcon: true,
+                onTap: () {},
+              ),
+            ],
+          );
+        }).toList(),
+      );
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      home: Material(child: buildCustomTable(
+        checkboxHorizontalMargin: _customCheckboxHorizontalMargin,
+        horizontalMargin: _customHorizontalMargin,
+      )),
+    ));
+
+    // custom checkbox padding
+    checkbox = find.byType(Checkbox).first;
+    padding = find.ancestor(of: checkbox, matching: find.byType(Padding));
+    expect(
+      tester.getRect(checkbox).left - tester.getRect(padding).left,
+      _customCheckboxHorizontalMargin,
+    );
+    expect(
+      tester.getRect(padding).right - tester.getRect(checkbox).right,
+      _customCheckboxHorizontalMargin,
+    );
+
+    // first column padding
+    padding = find.widgetWithText(Padding, 'Frozen yogurt').first;
+    cellContent = find.widgetWithText(Align, 'Frozen yogurt'); // DataTable wraps its DataCells in an Align widget
```

Comment:
super nit, period at the end: `// DataTable wraps its DataCells in an Align widget.`